### PR TITLE
Improve whitespace handling

### LINKED ISSUE #43: Modify parser's whitespace handling
Hello. I have found a bug with the current parser. Here is an example:

```
% cat .env
# tests
LF_BARE=hello\nworld
LF_DQ="hello\nworld"
LF_SQ='hello\nworld'

CRLF_BARE=hello\r\nworld
CRLF_DQ="hello\r\nworld"
CRLF_SQ='hello\r\nworld'

TRIM_BARE= hello world 
TRIM_DQ=" hello world "
TRIM_DQ2= " hello world " 
TRIM_SQ=' hello world '
TRIM_SQ2= ' hello world ' 

% cat example.ts
import {config} from 'https://deno.land/x/dotenv@v1.0.1/mod.ts';
console.log(config());

% deno run --allow-read=.env example.ts
{
  LF_BARE: "hello\\nworld",
  LF_DQ: "hello\nworld",
  LF_SQ: "hello\\nworld",
  CRLF_BARE: "hello\\r\\nworld",
  CRLF_DQ: "hello\\r\nworld",
  CRLF_SQ: "hello\\r\\nworld",
  TRIM_BARE: "hello world",
  TRIM_DQ: "hello world",
  TRIM_DQ2: "hello world",
  TRIM_SQ: "hello world",
  TRIM_SQ2: "hello world"
}

```

For all of the keys which begin with `TRIM_` (except for `TRIM_BARE`), the results are not what is expected.

If a value is not quoted, then I understand that trimming leading and trailing whitespace is desirable, but for quoted values, this should not be the case, since it makes it impossible to define values including whitespace at the ends.

I think this module is based on the Node `dotev` module, right? Here is a reference to its [parsing rules](https://github.com/motdotla/dotenv/blob/7301ac9be0b2c766f865bbe24280bf82586d25aa/README.md#rules):

> ### Rules
> 
> The parsing engine currently supports the following rules:
> 
> - `BASIC=basic` becomes `{BASIC: 'basic'}`
> - empty lines are skipped
> - lines beginning with `#` are treated as comments
> - empty values become empty strings (`EMPTY=` becomes `{EMPTY: ''}`)
> - inner quotes are maintained (think JSON) (`JSON={"foo": "bar"}` becomes `{JSON:"{\"foo\": \"bar\"}"`)
> - whitespace is removed from both ends of unquoted values (see more on [`trim`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/Trim)) (`FOO=  some value  ` becomes `{FOO: 'some value'}`)
> - single and double quoted values are escaped (`SINGLE_QUOTE='quoted'` becomes `{SINGLE_QUOTE: "quoted"}`)
> - single and double quoted values maintain whitespace from both ends (`FOO="  some value  "` becomes `{FOO: '  some value  '}`)
> - double quoted values expand new lines (`MULTILINE="new\nline"` becomes
> 
> ```
> {MULTILINE: 'new
> line'}
> ```

Also, I'm not sure how the original Node module handles CRLFs in quoted values, but (as you can see in the example) this module does not convert double-quoted CRLF (`\\r\\n`) to `\r\n` — only the LFs. Perhaps this can be changed as well.

### PR DESCRIPTION
**Pull request by jsejcksn**
Resolves #43

PTAL @pietvanzoen. I'll make some specific notes in one or more follow up comments.

### CODE REVIEW

File: mod.ts [hunk 1]
```diff
@@ -75,15 +81,19 @@ function parseFile(filepath: string) {
...
-function cleanQuotes(value: string = ""): string {
-  return value.replace(/^['"]([\s\S]*)['"]$/gm, "$1");
+function hasDoubleQuotes(str: string): boolean {
+  return /^"([\s\S]*)"$/.test(str);
 }
 
 function expandNewlines(str: string): string {
-  return str.replace("\\n", "\n");
+  return str.replaceAll("\\n", "\n");
 }
```
Comment (jsejcksn) [inline mod.ts:97]: Previously, this only replaced the first newline. Now it replaces all newlines.

File: README.md [hunk 2]
```diff
@@ -102,16 +102,17 @@ The parsing engine currently supports the following rules:
...
-- new lines are expanded if in double quotes (`MULTILINE="new\nline"` becomes
-- Variables that already exist in the environment are not overridden with `export: true`
+- new lines are expanded in double quoted values (`MULTILINE="new\nline"` becomes
 
 ```
 {MULTILINE: 'new
 line'}
 ```
 
+- Variables that already exist in the environment are not overridden with `export: true`
```
Comment (jsejcksn) [inline README.md:112]: This is no longer out of order.

**Comment by pietvanzoen** [comment]
Thanks for the work on all this @jsejcksn ! Looks good and will merge. 

I'll make it a major release as we're changing the trimming behaviour. But the change makes sense.

File: util.ts [hunk 3]
```diff
@@ -1,14 +1,6 @@
...
-export function compact(obj: any): object {
-  return Object.keys(obj).reduce((result: any, key) => {
-    if (obj[key]) {
-      result[key] = obj[key];
-    }
-    return result;
-  }, {});
+// deno-lint-ignore no-explicit-any
+export function compact(obj: Record<string, any>): Record<string, any> {
+  return Object.fromEntries(Object.entries(obj).filter(([, value]) => value));
```
[the remark this answers is not in the document]
Comment (jsejcksn) [reply util.ts:3]: @pietvanzoen What did you think about this?

[the remark this answers is not in the document]
Comment (pietvanzoen) [reply util.ts:3]: I think this is a good suggestion. Already merged but will implement before releasing.

[the remark this answers is not in the document]
Comment (jsejcksn) [reply util.ts:3]: This note is just for me before I resolve the comment conversation:

Change result:
https://github.com/pietvanzoen/deno-dotenv/blob/fca0b2eee5420962dae35d50e8763b1e069a58f3/util.ts#L2-L13

**Pull request merged**